### PR TITLE
Fix label attribution in token classification examples

### LINKED ISSUE #14043: Running `run_ner_no_trainer.py` with `--label_all_tokens` falsifies seqeval results
## Environment info

- `transformers` version: 4.11.3
- Platform: Linux-5.4.0-80-generic-x86_64-with-glibc2.17
- Python version: 3.8.11
- PyTorch version (GPU?): 1.9.0 (True)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: True
- Using distributed or parallel set-up in script?: False

## Who can help

Examples:

- maintained examples (not research project or legacy): @sgugger, @patil-suraj

## Description

Hi 🤗 team, not really an bug report, as the code is doing what it should, but not strictly a feature request either. 

When running `python run_ner_no_trainer.py` with `--label_all_tokens`, evaluating with `seqeval` (as default in the script) completely fudges the results. This is due to the fact that `seqeval` is entity based, which means that each sub-token in a "B-" labelled word  will be considered as a single entity. 

Example : 

Before tokenisation: 
```
O    B-PERS      I-PERS  I-PERS
The  Australian  Prime   minister
```
-> Number of entities: 1

After tokenisation with `label_all_tokens`:
 ```
O    B-PERS   B-PERS   I-PERS  I-PERS
The  Austral  ##ian    Prime   minister
```
-> Number of entities: 2

This can yield pretty important differences which may easily be overseen. I think it is worth adding a warning when using the script with this configuration.

### PR DESCRIPTION
**Pull request by sgugger**
# What does this PR do?

Currently, when a word is split into tokens in the token classification examples, all  tokens get the same label as the word. But if that label is a B-Xxx, only the first token should get it, the following ones should have an I-Xxx label. This PR addresses that.

Fixes #14043

### CODE REVIEW

**Review by patil-suraj — APPROVED** [review]
Thanks a lot for fixing this!
